pkg/kubelet/rkt/rkt.go

```diff
@@ -925,6 +968,34 @@ func (r *Runtime) preparePod(pod *api.Pod, pullSecrets []api.Secret) (string, *k
 		return "", nil, fmt.Errorf("failed to generate 'rkt run-prepared' command: %v", err)
 	}
 
+	runningPod := apiPodToruntimePod(uuid, pod)
+
+	if !kubecontainer.IsHostNetworkPod(pod) {
+		// Create a new network namespace for the pod
+		exec.Command("ip", "netns", "del", uuid).Run()
+		if err := exec.Command("ip", "netns", "add", uuid).Run(); err != nil {
+			return "", nil, fmt.Errorf("failed to create pod network namespace: %v", err)
+		}
+
+		// Set up networking with the network plugin
+		glog.V(3).Infof("Calling network plugin %s to setup pod for %s", r.networkPlugin.Name(), format.Pod(pod))
+		pluginPodID := buildContainerID(&containerID{uuid, pod.Name})
+		err = r.networkPlugin.SetUpPod(pod.Namespace, pod.Name, pluginPodID)
+		if err != nil {
+			r.cleanupPodNetwork(runningPod, false)
+			return "", nil, fmt.Errorf("failed to set up pod network: %v", err)
+		}
+
+		if r.configureHairpinMode {
+			if err = hairpin.SetUpContainerPath(netnsPathFromUuid(uuid), network.DefaultInterfaceName); err != nil {
+				glog.Warningf("Hairpin setup failed for pod %q: %v", format.Pod(pod), err)
+			}
+		}
+
+		// Run pod in the namespace we just created and set up
+		runPrepared = fmt.Sprintf("%s netns exec %s %s", r.ipPath, uuid, runPrepared)
```

Comment:
I found that `ip netns exec` breaks on some systems (e.g. my development laptop, running a fairly modern gentoo install):

```
$ sudo ip netns add foo
$ sudo ip netns exec foo rkt run --net=host docker://busybox --exec=ls -- /sys/fs/cgroup
image: using image from local store for image name coreos.com/rkt/stage1-coreos:1.4.0+gitb7589f9
image: using image from local store for url docker://busybox
stage1: error calling sd_pid_get_owner_uid: exec format error
$ sudo ip netns exec foo ls /sys/fs/cgroup
# nothing
$ ip -V
ip utility, iproute2-ss160111
```

`nsenter` didn't have that issue for me and `nsenter` is already depended on for `ExecInContainer`.

I can't reproduce this issue on CoreOS (`ip -V -> iproute2-ss150210`)

Ref: https://bugzilla.redhat.com/show_bug.cgi?id=882047

That's the reason I used `nsenter` for execing rather than `ip`. I don't know the state of this bug on other distros, but if you're confident it doesn't affect any real users (only the fake users running gentoo :) or if it's a kernel misconfiguration on my end or such, I'm happy to be persuaded. Otherwise, nsenter seems like the safer way to be sure this isn't an issue.